kms/asymmetric/samples.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"golang.org/x/net/context"
+	"google.golang.org/api/cloudkms/v1"
+	"log"
+	"math/big"
+
+	"golang.org/x/oauth2/google"
+)
+
+// [START kms_get_asymmetric_public]
+// Retrieve a public key from a saved asymmetric key pair on KMS
+func getAsymmetricPublicKey(client *cloudkms.Service, ctx context.Context, keyPath string) (interface{}, error) {
+	response, err := client.Projects.Locations.KeyRings.CryptoKeys.CryptoKeyVersions.
+		GetPublicKey(keyPath).Context(ctx).Do()
+	if err != nil {
+		return nil, err
+	}
+	if response == nil {
+		return nil, fmt.Errorf("no response from GetPublicKey")
+	}
+	keyBytes := []byte(response.Pem)
+	block, _ := pem.Decode(keyBytes)
+	publicKey, err := x509.ParsePKIXPublicKey(block.Bytes)
```

Comment:
instead of returning interface{} you can perhaps do

```
type pubKey struct {
   rsa rsa.PublicKey
   ecdsa ecdsa.PublicKey
}
```

```
switch v:= publicKey.(type) {
case *rsa.PublicKey: return &pubKey{rsa: v}, nil
case *ecdsa.PublicKey: return &pubKey{ecdsa: v}, nil
default: return nil, fmt.Errorf("unrecognized key format %T", v)
}
```